server/src/main/java/org/elasticsearch/rest/action/document/RestTermVectorsAction.java

```diff
@@ -43,14 +45,18 @@
  * TermVectorsRequest.
  */
 public class RestTermVectorsAction extends BaseRestHandler {
+    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(
+        LogManager.getLogger(RestTermVectorsAction.class));
+    static final String ENDPOINT_DEPRECATION_MESSAGE = "The _termvector" +
+        " endpoint is deprecated, use _termvectors (plural) instead.";
+
     public RestTermVectorsAction(Settings settings, RestController controller) {
         super(settings);
         controller.registerHandler(GET, "/{index}/{type}/_termvectors", this);
         controller.registerHandler(POST, "/{index}/{type}/_termvectors", this);
         controller.registerHandler(GET, "/{index}/{type}/{id}/_termvectors", this);
         controller.registerHandler(POST, "/{index}/{type}/{id}/_termvectors", this);
 
-        // we keep usage of _termvector as alias for now
         controller.registerHandler(GET, "/{index}/{type}/_termvector", this);
         controller.registerHandler(POST, "/{index}/{type}/_termvector", this);
         controller.registerHandler(GET, "/{index}/{type}/{id}/_termvector", this);
```

Comment:
should we use `registerWithDeprecatedHandler` instead?

Reply:
Thanks, I had overlooked this.